src/providers/list/utils/k8sGetProvidersByNamespace.ts

```diff
@@ -0,0 +1,19 @@
+import { ProviderModel, type V1beta1Provider } from '@kubev2v/types';
+import { k8sGet } from '@openshift-console/dynamic-plugin-sdk';
+
+/**
+ * Fetching providers list by namespace.
+ *
+ * @param {string} namespace to fetch providers by.
+ * @returns {Promise<V1beta1Provider[]>} providers list by namespace.
+ */
+export const k8sGetProvidersByNamespace = async (
+  namespace: string | undefined,
+): Promise<V1beta1Provider[]> => {
+  type K8sListResponse<T> = {
+    items: T[];
+  };
+  const providersList = await k8sGet({ model: ProviderModel, ns: namespace });
```

Comment:
only one time get? no watcher?

Reply:
There is a watcher in an upper level of usage. This utility method is called for non admin users in an interval of 20 seconds or when namespace is changed (via the `getInventoryByNamespace` method):
https://github.com/kubev2v/forklift-console-plugin/blob/dff568081a0043609884a52770e975dde131b851/src/providers/list/hooks/useProvidersInventoryList.ts#L97

Reply:
Yes, why? If u have a listener to inventory? You can filter by NS or change NS in the listener

Reply:
The getInventoryByNamespace function calls k8sGet for fetching the provider CR so both are done per interval.

Reply:
The `getInventoryByNamespace` function calls k8sGet for fetching the provider CR so both are done per interval